client/src/components/Comments/index.tsx

```diff
@@ -38,47 +40,57 @@ class Comments extends React.Component<ICommentsProps> {
         }
     }
 
-    public componentWillUnmount(): void {
-        this.props.resetComments();
-    }
-
     public getMoreComments = (): void => {
-        if (!this.props.allCommentsLoaded && this.props.postId) {
-            this.props.getComments(this.props.postId, this.props.commentsPage + 1);
+
+        if (!!this.props.postId) {
+            const commentStateForCurrentPost =
+                this.props.commentsPage.filter((info: { postId: string, page: number }) =>
+                    info.postId === this.props.postId);
+            // @ts-ignore
+            if (!this.props.allCommentsLoaded.includes(this.props.postId) && commentStateForCurrentPost.length) {
+                // @ts-ignore
```

Comment:
@ts-ignore - why?

Reply:
array methods want typeof this.props.postId === never
I can't do this, cause at that case has conflict